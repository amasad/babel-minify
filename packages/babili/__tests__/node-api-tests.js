jest.autoMockOff();

const babili = require("../src/index");

const sampleInput = `
function foo() {
  const bar = x(1);
  const baz = y(2);
  return z(bar, baz);
}
`;

describe("Babili Node API", () => {
  it("simple usage", () => {
    expect(babili(sampleInput).code).toMatchSnapshot();
  });

  it("throw on invalid options", () => {
    expect(() => babili(sampleInput, { foo: false, bar: true }).code).toThrow();
  });

  it("override default babili options", () => {
    const babiliOpts = { mangle: false };
    expect(babili(sampleInput, babiliOpts).code).toMatchSnapshot();
  });
});
