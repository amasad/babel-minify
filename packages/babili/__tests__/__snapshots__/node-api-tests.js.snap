// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babili Node API override default babili options 1`] = `"function foo(){const bar=x(1),baz=y(2);return z(bar,baz)}"`;

exports[`Babili Node API simple usage 1`] = `"function foo(){const a=x(1),b=y(2);return z(a,b)}"`;
